components/registry/org.wso2.carbon.registry.extensions/src/main/java/org/wso2/carbon/registry/extensions/handlers/WSDLMediaTypeHandler.java

```diff
@@ -363,63 +363,82 @@ public void put(RequestContext requestContext) throws RegistryException {
-                    wsdlPath = processWSDLImport(requestContext, wsdlProcessor, metadata, uri);
-                }
-                delete(tempFile);
-                if (wsdlPath != null) {
-                    onPutCompleted(path, Collections.singletonMap(uri, wsdlPath),
-                            Collections.<String>emptyList(), requestContext);
-                    requestContext.setActualPath(wsdlPath);
-                }
+            if (requestContext.getSourceURL() != null) {
+                requestContext.setResource(metadata);
+                String sourceURL = requestContext.getSourceURL();
+
+                wsdlProcessor = buildWSDLProcessor(requestContext);
+                String wsdlPath = processWSDLImport(requestContext, wsdlProcessor, metadata, sourceURL);
+
+                onPutCompleted(path, Collections.singletonMap(sourceURL, wsdlPath),
+                        Collections.<String>emptyList(), requestContext);
+                requestContext.setActualPath(wsdlPath);
+            } else {
+                try {
+                    Object resourceContent = metadata.getContent();
+                    byte[] resourceContentBytes;
+
+                    if (resourceContent == null) {
+                        resourceContentBytes = new byte[0];
+                    } else if (resourceContent instanceof byte[]) {
+                        resourceContentBytes = (byte[]) resourceContent;
+                    } else if (resourceContent instanceof String) {
+                        resourceContentBytes = RegistryUtils.encodeString((String) resourceContent);
+                    } else {
+                        String msg = "Unknown type for the content path: " + path + ", content type: " +
+                                resourceContent.getClass().getName() + ".";
+                        log.error(msg);
+                        throw new RegistryException(msg);
+                    }
+                    InputStream in = new ByteArrayInputStream(resourceContentBytes);
+
+                    File tempFile = File.createTempFile("wsdl", ".wsdl");
+
+                    BufferedOutputStream out = new BufferedOutputStream(new FileOutputStream(tempFile));
+                    byte[] contentChunk = new byte[1024];
+                    int byteCount;
+                    while ((byteCount = in.read(contentChunk)) != -1) {
+                        out.write(contentChunk, 0, byteCount);
+                    }
+                    out.flush();
+                    out.close();
+                    String uri = tempFile.toURI().toString();
+                    if (uri.startsWith("file:")) {
+                        uri = uri.substring(5);
+                    }
+                    while (uri.startsWith("/")) {
+                        uri = uri.substring(1);
+                    }
+                    uri = "file:///" + uri;
+                    String wsdlPath = null;
+                    if (uri != null) {
```

Comment:
will move the null check two lines upward. before  "uri = "file:///" + uri;" line